### PR TITLE
Reduce the padding overhead of unaligned MXFP8 GEMM shapes

### PR DESCRIPTION
**Pull request by b8zhong**
## Motivation

FlashInfer's CUTLASS MXFP8 GEMM needs N divisible by 32, so ragged output shards need a zero-padded weight and block scales. Today those are rebuilt on every forward, even though they depend only on constant weights.

## Modifications

Build them once in `process_weights_after_loading` and reuse them in the linear path.

Based on #32740; the two can merge independently.

## Accuracy Test

The padded tensors are bit-identical to what the per-forward path produced. 88 unit tests pass.

## Benchmarking and Profiling

`bench_gpu_time_with_cupti`, CUDA graphs, B300, n=1281 k=4096 (µs):

| m | before | after | speedup |
|---|---|---|---|
| 1 | 16.2 | 6.9 | 2.35x |
| 64 | 18.8 | 9.9 | 1.90x |
| 4096 | 35.9 | 24.7 | 1.45x |

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#running-unit-tests-adding-to-ci).

---
### CI States

Latest PR Test (Base): :x: [Run #30707229157](https://github.com/sgl-project/sglang/actions/runs/30707229157)
Latest PR Test (Extra): :x: [Run #30707229062](https://github.com/sgl-project/sglang/actions/runs/30707229062)

### CODE REVIEW

File: python/sglang/srt/layers/quantization/fp8.py [hunk 1]
```diff
@@ -769,7 +771,29 @@ def _process_mxfp8_linear_weight_scale(self, layer: Module) -> None:
...
+
+            # FlashInfer's CUTLASS MXFP8 GEMM requires N divisible by 32.  Pad
+            # here rather than in apply(): the inputs are constant weights, so
+            # per-forward padding re-materializes the whole padded weight on
+            # every token.
+            layer.weight_mxfp8_padded = None
+            if pad_rows:
+                layer.weight_mxfp8_padded = torch.cat(
+                    [weight, weight.new_zeros((pad_rows, k))], dim=0
+                ).contiguous()
```
Comment (mmangkad) [inline python/sglang/srt/layers/quantization/fp8.py:789]: `copy_or_rebind_param`?

**Review by mmangkad — APPROVED** [review]

**Comment by b8zhong** [comment]
No need for this PR, https://github.com/sgl-project/sglang/pull/32740
